core/txpool/celo_rates.go

```diff
@@ -0,0 +1,93 @@
+package txpool
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/contracts"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var (
+	unitRate = big.NewRat(1, 1)
+)
+
+func GetRatesForState(config *params.ChainConfig, state *state.StateDB) common.ExchangeRates {
+	stateCopy := state.Copy()
+	caller := &core.CeloBackend{
+		ChainConfig: config,
+		State:       stateCopy,
+	}
+
+	rates, err := caller.GetExchangeRates()
+	if err != nil {
+		log.Error("Error trying to get exchange rates.", "cause", err)
+	}
+	return rates
+}
+
+func ConvertCurrency(exchangeRates common.ExchangeRates, val1 *big.Int, currencyFrom *common.Address, currencyTo *common.Address) *big.Int {
+	goldAmount, err := contracts.ConvertCurrencyToGold(exchangeRates, val1, currencyFrom)
+	if err != nil {
+		log.Error("Error trying to convert from currency to gold.", "value", val1, "fromCurrency", currencyFrom)
+	}
+	toAmount, err := contracts.ConvertGoldToCurrency(exchangeRates, currencyTo, goldAmount)
+	if err != nil {
+		log.Error("Error trying to convert from gold to currency.", "value", goldAmount, "toCurrency", currencyTo)
+	}
+	return toAmount
+}
+
+func getRate(exchangeRates common.ExchangeRates, currency *common.Address) (*big.Rat, error) {
```

Comment:
We usually call this parameter `feeCurrency`, maybe use that for consistency.

Reply:
done